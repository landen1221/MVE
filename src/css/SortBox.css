.SortBox form {
  padding-top: 7px;
  display: inline;
}

.SortBox .MuiOutlinedInput-root {
  height: 3em;
  width: 10em;
  background-color: white;
  display: inline;
}

.SortBox .MuiInputBase-root {
  font-size: 0.75rem;
  margin: 10px;
  display: inline;
}

.SortBox h4 {
  display: inline;
  margin-top: 70px;
  margin-left: 0;
}

.SortBox .MuiTouchRipple-root {
  margin-top: 0;
}

.SortBox #clear {
  text-decoration: underline;
  margin-top: 0;
}

.SortBox .MuiPaper-root {
  margin: 5px 25px;
}

.SortBox .MuiButton-label {
  margin-top: 16px;
}

#filter {
  margin-top: 0px;
}

.SortBox .MuiInputBase-input {
  font-size: 15px;
  height: 0;
  width: 50%;
}
