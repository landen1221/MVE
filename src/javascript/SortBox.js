import "../css/SortBox.css";
import { makeStyles } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import TextField from "@material-ui/core/TextField";
import MenuItem from "@material-ui/core/MenuItem";
import grey from "@material-ui/core/colors/grey";
import { useEffect, useState } from "react";
import Button from "@material-ui/core/Button";

const useStyles = makeStyles((theme) => ({
  greyPaper: {
    backgroundColor: grey[200],
  },
}));

const SortBox = ({ dbName, storyList, setStoryList, originalStories }) => {
  const classes = useStyles();
  const gender = ["Male", "Female", "Other"];

  const initialValues = {
    gender: "",
    ageMin: "",
    ageMax: "",
    satisfied: "",
  };

  const [formData, setFormData] = useState(initialValues);

  function handleChange(evt) {
    const { name, value } = evt.target;

    setFormData((f) => ({
      ...f,
      [name]: value,
    }));
  }

  useEffect(() => {
    let filteredList = originalStories;
    for (let val in formData) {
      if (formData[val]) {
        if (isNaN(parseInt(formData[val]))) {
          filteredList = filteredList.filter(
            (story) => story[val] === formData[val]
          );
        }
        if (parseInt(formData["ageMin"]) > 0) {
          filteredList = filteredList.filter(
            (story) =>
              story["age"] >= parseInt(formData["ageMin"]) && story["age"]
          );
        }
        if (parseInt(formData["ageMax"]) > 0) {
          filteredList = filteredList.filter(
            (story) =>
              story["age"] <= parseInt(formData["ageMax"]) && story["age"]
          );
        }
      }
    }

    if (filteredList) {
      setStoryList(filteredList);
    }
  }, [formData, originalStories, setStoryList]);

  const clear = () => {
    setFormData({
      gender: "",
      ageMin: "",
      ageMax: "",
      satisfied: "",
    });
    setStoryList(originalStories);
  };

  const covidOrVaccine = (dbName) => {
    const covidSatisfied = ["No Big Deal", "Mild", "Moderate", "Severe"];
    const vaccineSatisfied = ["Yes", "No"];
    const data = dbName === "covid" ? covidSatisfied : vaccineSatisfied;
    return (
      <TextField
        id="select-satisfied"
        select
        label={dbName === "covid" ? "Intensity" : "Satisfied"}
        name="satisfied"
        variant="outlined"
        value={formData.satisfied}
        onChange={handleChange}
      >
        {data.map((val) => {
          return (
            <MenuItem key={val} value={val} name={val}>
              {val}
            </MenuItem>
          );
        })}
      </TextField>
    );
  };

  return (
    <div className={`SortBox ${classes.root}`}>
      <Paper elevation={2} className={classes.greyPaper}>
        <h4 id="filter">Filters: </h4>

        <form>
          <TextField
            id="select-gender"
            select
            label="Gender"
            name="gender"
            variant="outlined"
            value={formData.gender}
            onChange={handleChange}
          >
            {gender.map((val) => {
              return (
                <MenuItem key={val} value={val} name={val}>
                  {val}
                </MenuItem>
              );
            })}
          </TextField>
          {covidOrVaccine(dbName)}
          <TextField
            label="Min. Age"
            variant="outlined"
            name="ageMin"
            value={formData.ageMin}
            onChange={handleChange}
          />
          <TextField
            label="Max Age"
            variant="outlined"
            name="ageMax"
            value={formData.ageMax}
            onChange={handleChange}
          />

          <Button color="primary" id="clear" onClick={clear}>
            clear filters
          </Button>
        </form>
      </Paper>
    </div>
  );
};

export default SortBox;
