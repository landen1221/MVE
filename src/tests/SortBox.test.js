import React from "react";
import { render } from "@testing-library/react";
import SortBox from "../javascript/SortBox";
import { storyList } from "./globalTestVariables";

// Props: dbName, storyList, setStoryList, originalStories
// FIXME: How to utilize setStoryList
const dbName = "covid";
const originalStories = storyList;

it("renders without crashing", () => {
  render(
    <SortBox
      dbName={dbName}
      originalStories={originalStories}
      sotryList={storyList}
    />
  );
});
