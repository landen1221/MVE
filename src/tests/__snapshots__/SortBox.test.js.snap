// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SortBox undefined"
  >
    <div
      class="MuiPaper-root makeStyles-greyPaper-6 MuiPaper-elevation2 MuiPaper-rounded"
    >
      <h4
        id="filter"
      >
        Filters: 
      </h4>
      <form>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="select-gender"
            id="select-gender-label"
          >
            Gender
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="select-gender-label select-gender"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              id="select-gender"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              data-testid="gender"
              name="gender"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-9"
              >
                <span>
                  Gender
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="select-satisfied"
            id="select-satisfied-label"
          >
            Satisfied
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="select-satisfied-label select-satisfied"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              id="select-satisfied"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              name="satisfied"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-9"
              >
                <span>
                  Satisfied
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
          >
            Min. Age
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              name="ageMin"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-9"
              >
                <span>
                  Min. Age
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
          >
            Max Age
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              name="ageMax"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-9"
              >
                <span>
                  Max Age
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
          id="clear"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            clear filters
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
